openstf/model/train.py

```diff
@@ -73,29 +72,75 @@ def is_data_sufficient(data):
     return is_sufficient
 
 
-def train_model_for_specific_pj(
-    pj, context, retrain_young_models=False, compare_to_old=True
-):
+def train_model_pipeline(pj, context, retrain_young_models=False, compare_to_old=True):
     """Main function that controls the training process of one prediction job.
 
-    Here, on a high level, the input data is collected, features are applied and the
+    On a high level: first, the input data is collected, features are applied and the
     model is trained and stored. Additionally models for energy components are also
     trained if requested. Model is saved if better than the old model unless specified
     otherwise.
 
     Args:
         pj (dict): Prediction job
         context (openstf.task.utils.TaskContext): Task context for logging
-        retrain_young_models (bool): Retrain models younger then 7 days if True
+        retrain_young_models (bool): Retrain models younger then MAX_AGE_YOUNG_MODEL
+            days if True
         compare_to_old (bool): Compare new trained model to old trained model if True
 
     Returns:
-        Trained model (FIXME can be various datatypes at present)
+        None
     """
+
     # TODO Maybe this whole function can be replaced with a scikit learn pipeline?
+    model_trainer = create_model_for_specific_pj(pj, context, retrain_young_models)
+
+    if model_trainer is None:
+        return
+
+    context.perf_meter.checkpoint("model-creation")
+
+    split_input_data = preprocess_for_specific_pj(pj, context)
+
+    context.perf_meter.checkpoint("preprocessing")
+
+    train_for_specific_pj(model_trainer, split_input_data)
+
+    context.perf_meter.checkpoint("training")
+
+    split_predicted_data = predict_after_training_for_specific_pj(
+        pj,
+        model_trainer,
+        split_input_data,
+    )
+
+    context.perf_meter.checkpoint("predicting")
+
+    evaluate_new_model_for_specific_pj(
+        pj,
+        context,
+        model_trainer,
+        split_input_data,
+        split_predicted_data,
+        compare_to_old,
+    )
+
+    context.perf_meter.checkpoint("evaluation")
+
+
+def create_model_for_specific_pj(pj, context, retrain_young_models=False):
+    """Create model trainer and set hyperparams.
+
+    Args:
+        pj (dict): Prediction job
+        context (openstf.task.utils.TaskContext): Task context for logging
+        retrain_young_models (bool): Retrain models younger then MAX_AGE_YOUNG_MODEL
+            days if True. Defaults to False.
+
+    Returns:
+        MLModelType: type of model trainer
```

Comment:
Is this right? This function created a `model_trainer` I think?

Reply:
True, I copied return from ModelTrainerCreator(), but it should be from ModelTrainerCreator.create_model_trainer(), so I changed it.